dns/dns_updater.go

```diff
@@ -167,18 +185,39 @@ func (u *updater) Update(entries controller.IngressEntries) error {
 	return nil
 }
 
-func (u *updater) determineManagedRecordSets(rrs []*route53.ResourceRecordSet) []*route53.ResourceRecordSet {
+func (u *updater) flattenRoute53ResourceRecordSet(rrs []*route53.ResourceRecordSet) []record {
+	records := make([]record, 0)
```

Comment:
I think making a 0-length array of records is quite confusing here.

You can instead declare `var records []record`, and the `append` function will `make()` if it needs to